api/v1beta1/moduleimagesconfig_types.go

```diff
@@ -35,6 +36,18 @@ type ModuleImageSpec struct {
 	Image string `json:"image"`
 	// generation counter of the image config
 	Generation int `json:"generation"`
+
+	// Build contains build instructions, in case image needs building
+	// +optional
+	Build *Build `json:"build,omitempty"`
+
+	// Sign contains sign instructions, in case image needs signing
+	// +optional
+	Sign *Sign `json:"sign,omitempty"`
+
+	// ImageRepoSecret contains pull secret for the image's repo, if needed
+	// +optional
+	ImageRepoSecret *v1.LocalObjectReference `json:"imageRepoSecret,omitempty"`
```

Comment:
Let's move this field to the `ModuleImagesConfigSpec`? It is one per module anyway. I don't think we should replicate it in every entry.